pyproject.toml

```diff
@@ -0,0 +1,117 @@
+[build-system]
+requires = [
+    "setuptools",
+    "setuptools-scm",
+    "scikit-build-core >=0.4.3", 
+    "nanobind >=1.3.2"
+]
+build-backend = "scikit_build_core.build"
+
+[project]
+authors = [
+    {name = "Matthew A. Price"},
+    {name = "Jason D. McEwen"},
+    {name= "Contributors"}
+]
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Operating System :: OS Independent",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+]
+description = "Differentiable and accelerated spherical transforms with JAX"
+dependencies = [
+    "numpy>=1.20,<2",
+    "colorlog",
+    "pyyaml",
+    "jax>=0.3.13",
+    "jaxlib",
+    "torch",
+    "pyssht",
+    "healpy",
+    "ducc0",
+]
+dynamic = [
+    "version"
+]
+keywords = [
+]
+name = "s2fft"
+readme = ".pip_readme.rst"
+requires-python = ">=3.8"
+license.file = "LICENSE.txt"
+urls.homepage = "https://github.com/astro-informatics/s2fft"
+
+[project.optional-dependencies]
+docs = [
+    "sphinx>=5.0.0",
+    "nbsphinx-link>=1.3.0",
+    "pydata-sphinx-theme>=0.12.0",
+    "sphinx_toolbox>=2.15.0",
+    "sphinx-tabs>=3.2.0",
+    "sphinxcontrib-bibtex>=2.4.1",
+    "sphinx-git>=11.0.0",
+    "sphinxcontrib-texfigure>=0.1.3",
+    "sphinx-copybutton",
+    "sphinxemoji",
+    "sphinx_rtd_theme",
+    "ipython>=7.16.1",
+    "jupyter>=1.0.0",
+]
+plotting = [
+    "pyvista",
+    "trame",
+    "ipykernel",
+    "ipywidgets",
+]
+tests = [
+    "pytest",
+    "pytest-cov",
+    "codecov",
+    "so3",
+]
+
+[tool.scikit-build]
+# Protect the configuration against future changes in scikit-build-core
+minimum-version = "0.4"
+# Setuptools-style build caching in a local directory
+build-dir = "build/{wheel_tag}"
+# Build stable ABI wheels for CPython 3.12+
+wheel.py-api = "cp312"
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+sdist.include = ["s2fft/_version.py"]
+
+[tool.pytest.ini_options]
+addopts = "--color=yes -v"
+testpaths = [
+    "tests",
+]
+filterwarnings = [
+    "error",
+    "ignore::UserWarning",
+    "ignore::DeprecationWarning",
+    "ignore:FutureWarning",
+]
+
+[tool.setuptools]
+packages = ["s2fft"]
+write_to = "s2fft/_version.py"
+
+[tool.setuptools.dynamic]
+dependencies = {file = ["requirements/requirements-core.txt"]}
+optional-dependencies.docs = { file = ["requirements/requirements-docs.txt"] }
+optional-dependencies.plotting = { file = ["requirements/requirements-plotting.txt"] }
+optional-dependencies.tests = { file = ["requirements/requirements-tests.txt"] }
+
```

Comment:
```suggestion
```
This section is now redundant with dependencies included directly in file above. The corresponding files in `requirements` directory can also be removed.